src/jit/gentree.h

```diff
@@ -4887,6 +4899,165 @@ struct GenTreePutArgStk : public GenTreeUnOp
 #endif
 };
 
+#if !defined(LEGACY_BACKEND) && defined(_TARGET_ARM_)
+// Represent the struct argument: split value in register(s) and stack
+struct GenTreePutArgSplit : public GenTreePutArgStk
+{
+    unsigned gtNumRegs;
+
+    GenTreePutArgSplit(GenTreePtr op1,
+                       unsigned slotNum PUT_STRUCT_ARG_STK_ONLY_ARG(unsigned numSlots),
+                       unsigned     numRegs,
+                       bool         isHfa,
+                       bool         putIncomingArgArea = false,
+                       GenTreeCall* callNode           = nullptr)
+        : GenTreePutArgStk(GT_PUTARG_SPLIT,
+                           TYP_STRUCT,
+                           op1,
+                           slotNum PUT_STRUCT_ARG_STK_ONLY_ARG(numSlots),
+                           putIncomingArgArea,
+                           callNode)
+        , gtNumRegs(numRegs)
+    {
+        gtIsHfa = isHfa;
+        ClearOtherRegs();
+        ClearOtherRegFlags();
+    }
+
+    // First reg of struct is aloways given by gtRegNum.
+    // gtOtherRegs holds the other reg numbers of struct.
+    // HFA args is not yet handled.
+    regNumberSmall gtOtherRegs[MAX_REG_ARG - 1];
+
+    // GTF_SPILL or GTF_SPILLED flag on a multi-reg struct node indicates that one or
+    // more of its result regs are in that state.  The spill flag of each of the
+    // return register is stored in the below array.
+    unsigned gtSpillFlags[MAX_REG_ARG];
+
```

Comment:
This looks the same as the `FEATURE_MULTIREG_RET` code in GenTreeCall. Can you adopt the recent change I made there, converting gtSpillFlags from an array of 4 32-bit unsigned elements (for arm32), namely 16 bytes, to a single 8-bit `unsigned char`? The key is that only data you need to store is 2 bits, for GTF_SPILL and GTF_SPILLED.